libs/libc/misc/lib_crc32.c

```diff
@@ -18,110 +18,127 @@
- * as "+1") results in the MSB being 1
- *
- * Note that the usual hardware shift register implementation, which is what we're using
- * (we're merely optimizing it by doing eight-bit chunks at a time) shifts bits into the
- * lowest-order term.  In our implementation, that means shifting towards the right.  Why
- * do we do it this way?  Because the calculated CRC must be transmitted in order from
- * highest-order term to lowest-order term.  UARTs transmit characters in order from LSB
- * to MSB.  By storing the CRC this way we hand it to the UART in the order low-byte to
- * high-byte; the UART sends each low-bit to hight-bit; and the result is transmission bit
- * by bit from highest- to lowest-order term without requiring any bit shuffling on our
- * part.  Reception works similarly
+ * Note that we take it "backwards" and put the highest-order term in the
+ * lowest-order bit. The X^32 term is "implied"; the LSB is the X^31 term,
+ * etc. The X^0 term (usually shown as "+1") results in the MSB being 1
+ *
+ * Note that the usual hardware shift register implementation, which is what
+ * we're using (we're merely optimizing it by doing eight-bit chunks at a
+ * time) shifts bits into the lowest-order term.  In our implementation, that
+ * means shifting towards the right.  Why do we do it this way?  Because the
+ * calculated CRC must be transmitted in order from
+ * highest-order term to lowest-order term.  UARTs transmit characters in
+ * order from LSB to MSB.  By storing the CRC this way we hand it to the UART
+ * in the order low-byte to high-byte; the UART sends each low-bit to
+ * hight-bit; and the result is transmission bit by bit from highest- to
+ * lowest-order term without requiring any bit shuffling on our part.
+ * Reception works similarly
  *
  * The feedback terms table consists of 256, 32-bit entries.  Notes
  *
- * - The table can be generated at runtime if desired; code to do so is shown later.  It
- *   might not be obvious, but the feedback terms simply represent the results of eight
- *   shift/xor operations for all combinations of data and CRC register values
+ * - The table can be generated at runtime if desired; code to do so is shown
+ *   later.  It might not be obvious, but the feedback terms simply represent
+ *   the results of eight shift/xor operations for all combinations of data
+ *   and CRC register values
  *
- * - The values must be right-shifted by eight bits by the updcrc logic; the shift must
- *   be u_(bring in zeroes).  On some hardware you could probably optimize the shift in
- *   assembler by using byte-swap instructions polynomial $edb88320
- ************************************************************************************************/
+ * - The values must be right-shifted by eight bits by the updcrc logic; the
+ *   shift must be u_(bring in zeroes).  On some hardware you could probably
+ *   optimize the shift in assembler by using byte-swap instructions
+ *   polynomial $edb88320
```

Comment:
@jinliangli from time to time some people face issues with CRC32 on NuttX (because the results are not the expected value), maybe we should include a Documentation/ about it or modify the CRC32 to use the Normal CRC32 Polynomial (looking this table I see we are using the Reversed Polynomial: https://en.wikipedia.org/wiki/Cyclic_redundancy_check#Polynomial_representations )